Cargo.toml

```diff
@@ -19,5 +19,9 @@ serde = "0.8"
 uuid = {version = "0.3", features = ["v4"]}
 fnv = "1.0.3"
 
+[target.'cfg(target_os = "linux")'.dependencies]
+mio = "0.6.1"
+fnv = "1.0.3"
```

Comment:
Hm, OK... I don't have experience with that, but it certainly does sound reasonable :-)

That calls for a separate PR though I'd say, as it's not really related to the changes here?